packages/fetchai/skills/aries_alice/handlers.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains the handlers for the aries_alice skill."""
+
+import base64
+import binascii
+import json
+from typing import Dict, Optional, cast
+from urllib.parse import urlparse
+
+from aea.configurations.base import ProtocolId, PublicId
+from aea.mail.base import Envelope, EnvelopeContext
+from aea.protocols.base import Message
+from aea.protocols.default.message import DefaultMessage
+from aea.skills.base import Handler
+
+from packages.fetchai.protocols.http.message import HttpMessage
+from packages.fetchai.protocols.http.serialization import HttpSerializer
+
+HTTP_CONNECTION_PUBLIC_ID = PublicId("fetchai", "http_client", "0.1.0")
+
+HTTP_PROTOCOL_PUBLIC_ID = PublicId("fetchai", "http", "0.1.0")
+DEFAULT_ADMIN_HOST = "127.0.0.1"
+DEFAULT_ADMIN_PORT = 8031
+
+
+class AriesDemoDefaultHandler(Handler):
+    """This class represents alice's handler for default messages."""
+
+    SUPPORTED_PROTOCOL = DefaultMessage.protocol_id  # type: Optional[ProtocolId]
+
+    def __init__(self, **kwargs):
+        """Initialize the handler."""
+        self.admin_host = kwargs.pop("admin_host", DEFAULT_ADMIN_HOST)
+        self.admin_port = kwargs.pop("admin_port", DEFAULT_ADMIN_PORT)
+
+        super().__init__(**kwargs)
+
+        self.admin_url = "http://{}:{}".format(self.admin_host, self.admin_port)
+
+        self.kwargs = kwargs
+        self.handled_message = None
+
+    def _admin_post(self, path: str, content: Dict = None):
+        # Request message & envelope
+        request_http_message = HttpMessage(
+            performative=HttpMessage.Performative.REQUEST,
+            method="POST",
+            url=self.admin_url + path,
+            headers="",
+            version="",
+            bodyy=b"" if content is None else json.dumps(content).encode("utf-8"),
+        )
+        context = EnvelopeContext(connection_id=HTTP_CONNECTION_PUBLIC_ID)
+        envelope = Envelope(
+            to="Alice_ACA",
```

Comment:
I would say this should be `self.admin_url`.

Reply:
Done